elastic/logstash
# support environment variable in `if` statement

## Release notes


Support Environment Variable in conditionals

## What does this PR do?



This PR substitutes environment variable in config string before parsing it to compile a syntax graph

## Why is it important/What is the impact to the user?



This PR allows users to use environment variable in condition, eg. `if [app] == "${ENV_VARIABLE}"` 

## Checklist




## Author's Checklist



## How to test this PR locally


```
input {
	generator {
		lines => ['{"app": "foobar"}']
	}
}
filter {
	json {
		source => "message"
	}
	if [app] == "${ENV_VARIABLE}" {
		mutate { add_tag => ["some_tag"] }
	}
}
output {
	stdout { codec => rubydebug }
}
```

run logstash with above config should see `tag` 
$ ENV_VARIABLE=foobar bin/logstash 

## Related issues


Fixed: #5115
## Use cases



## Screenshots



## Logs
